src/orchestrator.ts

```diff
@@ -595,16 +595,29 @@ export class Orchestrator {
         state: HistoryEvent[],
         name: string,
         instanceId: string)
-        : SubOrchestrationInstanceCreatedEvent {
-        const returnValue = name
-            ? state.filter((val: HistoryEvent) => {
+        : SubOrchestrationInstanceCreatedEvent | undefined {
+
+        if (name) {
+            const matches = state.filter((val: HistoryEvent) => {
                 return val.EventType === HistoryEventType.SubOrchestrationInstanceCreated
                     && (val as SubOrchestrationInstanceCreatedEvent).Name === name
-                    && (val as SubOrchestrationInstanceCreatedEvent).InstanceId === instanceId
                     && !val.IsProcessed;
-            })[0]
-            : undefined;
-        return returnValue as SubOrchestrationInstanceCreatedEvent;
+            });
+
+            if (matches.length > 0) {
+                // Grab the first unprocessed sub orchestration with the same name.
```

Comment:
We've already filtered the list of matches by name. Should the comment say "...with the same instance ID"?

Reply:
Substantially simplified the logic here.